tfx/components/example_gen/utils.py

```diff
@@ -397,6 +409,17 @@ def _retrieve_latest_span_version(
                                              is_match_span, is_match_date,
                                              is_match_version))
 
+    # Check parsed span number against range config, if it exists.
+    if range_config:
+      if range_config.HasField('static_range'):
+        if (match_span_int < range_config.static_range.start_span_number or
+            match_span_int > range_config.static_range.end_span_number):
+          # If span not in static range, skip to next file path.
+          continue
```

Comment:
for this, we don't want to scan the files, we just want to fill the span (with width) into the span spec or date spec, and directly access that span, if not available, just error

Reply:
@hughmiao a couple edge cases to consider. if range_config specifies span number of 1 and split pattern is `span{SPAN}/*`, if we did a naive substitution (just replace spec with int), this would fail for a filepath like `span0001/data`. May need some discussion on whether this behavior is intended or not.

Reply:
additionally, since we are doing replacement and not filtering, should we also add a RangeConfig for date logic? it seems like our range_config logic in substituting won't allow users to process specific dates?